### PR TITLE
[1.19.x] Add supplier to FlowerBlock so it works with custom MobEffects

### PR DESCRIPTION
**Pull request by Minecraftschurli**

### CODE REVIEW

**Comment by ChampionAsh5357** [comment]
This shouldn't be an issue anymore since mob effects are registered before blocks in the vanilla registry order.

**Comment by Minecraftschurli** [comment]
Ok then I'll close this and just open the 1.18.x one (because it definitely is an issue there)

**Comment by bconlon1** [comment]
This is able to be reopened as mob effects are actually registered after blocks, not before.

File: patches/minecraft/net/minecraft/world/level/block/FlowerBlock.java.patch [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
++   /** @deprecated FORGE: Use supplier version instead */
++   @Deprecated
+    public FlowerBlock(MobEffect p_53512_, int p_53513_, BlockBehaviour.Properties p_53514_) {
+       super(p_53514_);
+       this.f_53508_ = p_53512_;
+@@ -23,6 +_,7 @@
+       } else {
+          this.f_53509_ = p_53513_ * 20;
+       }
++      this.suspiciousStewEffectSupplier = null;
```
Comment (ChampionAsh5357) [inline patches/minecraft/net/minecraft/world/level/block/FlowerBlock.java.patch]: This should either return the mob effect supplied in the constructor or, to support registry replacement, a delegate similar to `LiquidBlock` or just by calling `RegistryObject#of`.

File: patches/minecraft/net/minecraft/world/level/block/FlowerBlock.java.patch [hunk 2]
```diff
@@ -0,0 +1,41 @@
+--- a/net/minecraft/world/level/block/FlowerBlock.java
++++ b/net/minecraft/world/level/block/FlowerBlock.java
+@@ -15,6 +_,17 @@
+    private final MobEffect f_53508_;
+    private final int f_53509_;
+ 
++   private final java.util.function.Supplier<MobEffect> suspiciousStewEffectSupplier;
++
++   public FlowerBlock(java.util.function.Supplier<MobEffect> suspiciousStewEffectSupplier, int p_53513_, BlockBehaviour.Properties p_53514_) {
```
Comment (ChampionAsh5357) [inline patches/minecraft/net/minecraft/world/level/block/FlowerBlock.java.patch]: Needs a forge.exc entry if you are going to use the param names.

**Review by ChampionAsh5357 — CHANGES_REQUESTED** [review]

**Review by ChampionAsh5357 — APPROVED** [review]

**Pull request merged**